utils/java_properties.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2023 Red Hat, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+const (
+	javaPropertyValidKeyRegExp = "^[a-zA-Z$_][a-zA-Z0-9$_\\.\\-]*$"
+)
+
+var _ JavaProperties = &javaProperties{}
+var validKeyRegExp = regexp.MustCompile(javaPropertyValidKeyRegExp)
```

Comment:
Doesn't Golang have a properties api already in place?
https://pkg.go.dev/github.com/magiconair/properties
we could use it instead implement a new one,

Reply:
That's nice! Mine is better and lightweight. :D

But I guess it's a better option to use one that's better tested, I think. And since they don't use any other modules, I believe we can do it: https://github.com/magiconair/properties/blob/main/go.mod